File: Easy/665 - Non-decreasingArray/Solution.cs
// Question 665. Non-decreasing Array (https://leetcode-cn.com/problems/non-decreasing-array/)

/// <summary>
/// 执行用时：152 ms, 在所有 C# 提交中击败了 31.43% 的用户
/// 内存消耗：31.6 MB, 在所有 C# 提交中击败了 38.57% 的用户
/// </summary>
public class Solution 
{
    public bool CheckPossibility(int[] nums) 
    {
        bool elementModified = false;

        int N = nums.Count();
        for (int i = 0; i < N - 1; i++)
        {
            int x = nums[i];
            int y = nums[i + 1];
            if (x > y)
            {
                if (!elementModified)
                {
                    elementModified = true;

                    if (i > 0 && y < nums[i - 1])
                    nums[i + 1] = x;
                }
                else
                {
                    return false;
                }
            }
        }

        return true;
    }
}

File: Easy/665 - Non-decreasingArray/solution.ts
// Question 665. Non-decreasing Array (https://leetcode-cn.com/problems/non-decreasing-array/)

/**
 * 执行用时：120 ms, 在所有 TypeScript 提交中击败了 8.00% 的用户
 * 内存消耗：40.5 MB, 在所有 TypeScript 提交中击败了 100.00% 的用户
 *
 * @param {number[]} nums
 * @returns {boolean}
 */
function checkPossibility(nums: number[]): boolean {
  let elementModified: boolean = false;

  for (let index = 0; index < nums.length - 1; index++) {
    const x = nums[index];
    const y = nums[index + 1];

    if (x > y) {
      if (elementModified) {
        return false;
      }
      elementModified = true;
      if (index > 0 && nums[index - 1] > y) {
        nums[index + 1] = x;
      }
    }
  }

  return true;
}
